#The scripts used in this project

For sRNA processing
    Code01.Low_Quality_filter.pl (remove the sRNA reads with low quality). 
    
    Code02.PolyA_trim.pl (remove the polyA/T reads).

For DR sRNA identification
  Code03.transposition_matrix.pl (transpose expression matrix). 
  
  Code04.Paired-T-test.R (used to perform paired-t-test). 
  
  Code05.eMatrix2biomarker.pl and Code06.Biomarker2hmp.pl (construct hmpfiles for sRNA expression). 

For sRNA-gene correlation
  Code07.correlation_coefficient.pl (used to calculate the correlation coefficient--R was needed). 

For eQTL identification
  Code08.candidate_eQTL_find.pl (identify the candidate eQTLs). 
  
  Code09.LD_treatment.pl (filter the candidate eQTL throuth LD and p-value). 
  
  Code10.arrange_joint_effect.pl (arrange the joint effect result of candidate eQTLs). 
  
  Code11.major_eQTL.pl(get the major eQTL information). 
  
  Code12.SNPs_used_for_JE_calculation.pl (arrange the format of SNPs used to calculate the joint effects). 

For IR identification
  Code13.SplitGenome.pl (split genome into the given lengths. The splited fragments will be further blast against itself by blastn at NCBI). 
  
  Code14.Treat_blastn_output.pl (arrange the output of the blast for further analyse). 
  
  Code15.IRfinder.pl (identify the IR region and give the IR ratio). 
   
For figure drawing
  Code16.LD_figure.pl (used to draw figure 2c). 
  
  Code17.Maize_V4_eQTL_map_drawing2.pl (used to draw figure 2a). 
  
  Code18.Maize_V4_eQTL_map_drawing.pl (used to draw Supplementary Results Fig. 4a).
  
  Code19.mhtplot.pl (used to draw Manhattan plots in Supplementary Results Fig. 3f, 3h). 
  
If there are any questions about using these scripts/code, please contact the authors at mingqiudai@mail.hzau.edu.cn.
