#The scripts used in this project

1. For sRNA processing:
   1) Remove 3' adaptor in reads.remove reads without 3' adaptor.
   2) Remove reads with 5' adaptor remove length < 18 && len > 30.

          cutadapt -a TGGAATTCTCGGGTGCCAAGG -o  INPUT --discard-untrimmed -m 18 -M 30 -e 0.05 -O 14 --no-indels OUTPUT 
          cutadapt -g GTTCAGAGTTCTACAGTCCGACGATC -o OUTPUT --discard-trimmed -e 0 -O 6 --no-indels INPUT 
        
   3) Remove low quality reads. (Q20 < 50% && N < 10);
    Use custom perl script Low_Quality_filter.pl
   4) Remove poly A/T (AT > 80% or poly A(T) > 30 %).
    Use custom perl script PolyA_trim.pl

        Code01.Low_Quality_filter.pl (remove the sRNA reads with low quality). 
        Code02.PolyA_trim.pl (remove the polyA/T reads).

   5) Size filtering and remove rRNA,tRNA,snoRNA and snRNA collected from databases below
                tRNA : GtRNAdb (http://gtrnadb2009.ucsc.edu/download.html)
		plant snoRNA database(http://bioinf.scri.sari.ac.uk/cgi-bin/plant_snorna/home/)
		rRNA:  silva(http://www.arb-silva.de/)
		all rRNA,tRNA,snoRNA,snRNA in rfam12.0(http://rfam.xfam.org/)
   6) sRNA reads mapped to genome by using bowtie with 0 mismatch allowed
        
2. For DR sRNA identification
  The expression matrix of sRNAs/ sRNA clusters were arranged by bash and perl command lines. For convenient, a perl script (Code03) was used to perform matrix transformation.
Paired-T-test between sRNAs/sRNA clusters expression unter WW (Well-Watered) and DS (Drought-Stressed) were performed by using a simple R script (Code04): Rscript Paired-T-test.R eMatrixWW eMatrixDS > OUTPUT

    Code03.transposition_matrix.pl (transpose expression matrix).
    Code04.Paired-T-test.R (used to perform paired-t-test).
    
  BiomarkerGWAS was performed by using TASSEL5, weight MLM model (PCA+K), see the GWAS part in eQTL identification below.   
  The hmpfiles used in biomarkerGWAS were treated from sRNAs/ sRNA clsters expression matrix by using two custom perl scripts (Code05 and Code06).

    perl eMatrix2biomarker.pl eMatrix > OUPUT
    perl Biomarker2hmp.pl INPUT(OUTPUT of Code05) > OUTPUT  
    Code05.eMatrix2biomarker.pl
    Code06.Biomarker2hmp.pl (construct hmpfiles for sRNA expression). 

3. For sRNA-gene correlation
  Regression analyse was performed by using custom script Code07. Perl correlation_coefficient.pl eMatrix_sRNA eMatrix_gene list > OUTPUT
The maize inbred of eMatrix_sRNA and gene eMatrix_gene should be the same and sored in same order. The perl CPAN module Statistics::R and R should be intalled in advance. 

    Code07.correlation_coefficient.pl (used to calculate the correlation coefficient--R was needed). 

4. For eQTL identification
  GWAS of DR sRNAs/sRNA clusters were performed by using TASSEL5, weighted MLM(PCA+K) was used.

    run_pipeline.pl -Xmx5g -fork1 -h SNPFILE -filterAlign -filterAlignMinFreq 0.05 -fork2 -r TRAITFILE -fork3 -r PCAFILE -fork4 -k KINSHIPFILE -combine5 -input1 -input2 -input3  -intersect -combine6 -input5 -input4 -mlm -mlmVarCompEst P3D -mlmCompressionLevel None -export OUTPUT -runfork1 -runfork2 -runfork3  -runfork4" ; done

  Both PCA (PC1~PC3) and Kinship(K) were calculated by TASSEL5 using all 1.23M SNPs.

    run_pipeline.pl -Xmx150g -fork1 -importGuess SNPFILE -PrincipalComponentsPlugin -covariance true -endPlugin -export OUTPUT -runfork1
    run_pipeline.pl -Xmx150g -fork1 -importGuess SNPFILE -KinshipPlugin -method Centered_IBS -endPlugin -export OUTPUT -exportType SqrMatrix

  The output file of GWAS were futher arranged to two types files, and these files were used as the input file of custom script(Code08) file to get candidate eQTLs
  File1 :
      gene/sRNA_id    SNP_id  SNP_chr SNP_loci    p-value
  File2 :
      SNP_id  SNP_loci-1  SNP_loci    gene/sRNA_id    p-value
  Commandline : perl candidate_eQTL_find.pl File1 File2 > OUTPUT
  the software bedtools was needed in this script.
    
    Code08.candidate_eQTL_find.pl (identify the candidate eQTLs). 
    
  Then all candidate eQTLs were further classified to major eQTL region or satellite eQTL region according to the p-value of lead SNP or joint effect (when p-values are equal) 
The LD between each lead SNPs were calculated by TASSEL5 and 1st,2nd,7th,8th,14th columns were used as the Input for Code09. After this step, the candidate eQTL in LD shows a lower Lead SNP p-value were defined as major eQTL.

    run_pipeline.pl -Xmx50G -fork1 -h LEAD_SNP_FILE -ld -ldType All -export OUTPUT
    Code09.LD_treatment.pl (filter the candidate eQTL throuth LD and p-value). 

  If lead SNPs of two candidate eQTLs in LD had same p-value, the joint effect of these candidate eQTLs were calculated by custom perl script(Code10-12). The candidate eQTL with higher joint effect was defined as major eQTL and other were defined as satellite eQTL.
    
    Code10.arrange_joint_effect.pl (arrange the joint effect result of candidate eQTLs). 
    Code11.major_eQTL.pl(get the major eQTL information). 
    Code12.SNPs_used_for_JE_calculation.pl (arrange the format of SNPs used to calculate the joint effects). 

5. For IR identification
  The maize V4 genome sequence were split to series 100kb fregments with a 50kb step.
    
    Code13.SplitGenome.pl (split genome into the given lengths. The splited fragments will be further blast against itself by blastn at NCBI).
    perl SplitGenome.pl GENOME_SEQUENCE 100000 50000 > OUTPUT_DIR
    
  And all these fregments were blast to itself with the parameter below: 
    
    makeblastdb -in INPUT_SEQ -dbtype nucl -parse_seqids -out INDEX
    blastn -query INPUT_SEQ -db INPUT_INDEX -evalue 1e-30 -outfmt 6|awk '($7-$8)*($9-$10)<0'|awk '$8<=$9' > OUTPUT
    
  Then the results of blast were further treated with custom script (Code14) and used for IR identification (Code15).
     
    Code14.Treat_blastn_output.pl (arrange the output of the blast for further analyse).
    Code15.IRfinder.pl (identify the IR region and give the IR ratio). 


6. For Fst and XPCLR calculation
  To calculate the Fst value, the hmp format SNPfiles were transformed to VCF format by using TASSEL5. Then Fst value was calculated by vcftools. For convenient, a custom python script (Code20) was used to get the subset of hmpfile according a given inbred list.
    
    run_pipeline.pl -Xmx10G -fork1 -h INPUT_HMP_FILE -export OUTPUT_VCF_FILE -exportType VCF 
    vcftools --vcf VCF_FILE --weir-fst-pop TEMP_POP_LIST --weir-fst-pop TST_POP_LIST --fst-window-size 20000 --fst-window-step 10000 --out OUTPUT (Fst_AVERAGE)
    vcftools --vcf VCF_FILE --weir-fst-pop TEMP_POP_LIST --weir-fst-pop TST_POP_LIST --out OUTPUT (Fst_SINGLE_POINT)
  To calculate the XPCLR score, the hmp format SNPfiles were transformed to the accecptable input format by two python scripts(Code21 and Code22). The genetic distance of SNPs were collected from previous study, and arranged to a fitable format for Code22.
  
    cat INPUT_HMP_FILE|grep -v "rs"|awk '{print $3"\t"$4-1"\t"$4"\t"$1"\t"$2}' > OUTPUT_HMPFILE.bed
    intersectBed -a OUTPUT_HMPFILE.bed -b Genetic_Distance_FILE -wa -wb >OUTPUTFILE (Input for Code 22)
    python hmp2xpclr_geno.py INPUT_HMP_FILE > OUTPUT_GENO_FILE
    python hmp2xpclr_snp.py INPUT_FILE > OUTPUT (Inputfile for XPCLR)
    XPCLR -xpclr INPUT_GENO_FILE1 INPUT_GENO_FILE2 INPUT_SNP_FILE RESULT -w1 0.005 600 10000 8 -p0 0.95 

7.The trait diffenence tests bewtween inbreds with different alleles were performed by custom python script (Code23). 
    
    python Genotype_Trait_Test.py HMP_FILE eMATRIX_FILE SNP_Trait_LIST > OUTPUT 
    OUTPUT format :
    SNP_id	Trait	Allele1	Number_of_Allele1	Trait_Average_value_of_Allele1	Trait	Allele2	Number_of_Allele2	Trait_Average_value_of_Allele2	P-value

8. For figure drawing
    
    Code16.LD_figure.pl (used to draw figure 2c). 
    Code17.Maize_V4_eQTL_map_drawing2.pl (used to draw figure 2a). 
    Code18.Maize_V4_eQTL_map_drawing.pl (used to draw Supplementary Results Fig. 4a).
    Code19.mhtplot.pl (used to draw Manhattan plots in Supplementary Results Fig. 3f, 3h). 
  
If there are any questions about using these scripts/code, please contact the authors at mingqiudai@mail.hzau.edu.cn.
