#The scripts used in this project:

For sRNA processing:
Low_Quality_filter.pl (remove the sRNA reads with low quality)
PolyA_trim.pl (remove the polyA/T reads)

For DR sRNA identification:
Paired-T-test.R (used to perform paired-t-test)
eMatrix2biomarker.pl and Biomarker2hmp.pl (construct hmpfiles for sRNA expression) transposition_matrix.pl (transpose expression matrix)

For sRNA-gene correlation:
correlation_coefficient.pl (used to calculate the correlation coefficient--R was needed)

For eQTL identification:
  candidate_eQTL_find.pl (identify the candidate eQTLs)
  SNPs_used_for_JE_calculation.pl (arrange the format of SNPs used to calculate the joint effects)
  LD_treatment.pl (filter the candidate eQTL throuth LD and p-value)
  arrange_joint_effect.pl (arrange the joint effect result of candidate eQTLs)
  major_eQTL.pl(get the major eQTL information)
  
For IR identification:
   SplitGenome.pl (split genome into the given lengths. The splited fragments will be further blast against itself by blastn at NCBI)
   Treat_blastn_output.pl (arrange the output of the blast for further analyse)
   IRfinder.pl (identify the IR region and give the IR ratio)
   
For figure drawing:
  Maize_V4_eQTL_map_drawing2.pl (used to draw figure 2a)
  LD_figure.pl (used to draw figure 2c)
  mhtplot.pl (used to draw Manhattan plots in Supplementary Results Fig. 3f, 3h)
  Maize_V4_eQTL_map_drawing.pl (used to draw Supplementary Results Fig. 4a)

If there are any questions about using these scripts/code, please contact the authors at mingqiudai@mail.hzau.edu.cn.
