#The scripts used in this project

For sRNA processing:
1) Remove 3' adaptor in reads.remove reads without 3' adaptor.
2) Remove reads with 5' adaptor remove length < 18 && len > 30.

        cutadapt -a TGGAATTCTCGGGTGCCAAGG -o  INPUT --discard-untrimmed -m 18 -M 30 -e 0.05 -O 14 --no-indels OUTPUT 
        cutadapt -g GTTCAGAGTTCTACAGTCCGACGATC -o OUTPUT --discard-trimmed -e 0 -O 6 --no-indels INPUT 
        
3) Remove low quality reads. (Q20 < 50% && N < 10);
    Use custom perl script Low_Quality_filter.pl
4) Remove poly A/T (AT > 80% or poly A(T) > 30 %).
    Use custom perl script PolyA_trim.pl

        Code01.Low_Quality_filter.pl (remove the sRNA reads with low quality). 
        Code02.PolyA_trim.pl (remove the polyA/T reads).

5) Size filtering and remove rRNA,tRNA,snoRNA and snRNA collected from databases below
        tRNA : GtRNAdb (http://gtrnadb2009.ucsc.edu/download.html)
		plant snoRNA database(http://bioinf.scri.sari.ac.uk/cgi-bin/plant_snorna/home/)
		rRNA:  silva(http://www.arb-silva.de/)
		all rRNA,tRNA,snoRNA,snRNA in rfam12.0(http://rfam.xfam.org/)
6) sRNA reads mapped to genome by using bowtie with 0 mismatch allowed
        
For DR sRNA identification
    
    Code03.transposition_matrix.pl (transpose expression matrix).
    Code04.Paired-T-test.R (used to perform paired-t-test). 
    Code05.eMatrix2biomarker.pl and Code06.Biomarker2hmp.pl (construct hmpfiles for sRNA expression). 

For sRNA-gene correlation
    
    Code07.correlation_coefficient.pl (used to calculate the correlation coefficient--R was needed). 

For eQTL identification
GWAS of DR sRNAs/sRNA clusters were performed by using TASSEL5, weighted MLM(PCA+K) was used.

    run_pipeline.pl -Xmx5g -fork1 -h SNPFILE -filterAlign -filterAlignMinFreq 0.05 -fork2 -r TRAITFILE -fork3 -r PCAFILE -fork4 -k KINSHIPFILE -combine5 -input1 -input2 -input3  -intersect -combine6 -input5 -input4 -mlm -mlmVarCompEst P3D -mlmCompressionLevel None -export OUTPUT -runfork1 -runfork2 -runfork3  -runfork4" ; done

Both PCA (PC1~PC3) and Kinship(K) were calculated by TASSEL5 using all 1.23M SNPs.

    run_pipeline.pl -Xmx150g -fork1 -importGuess SNPFILE -PrincipalComponentsPlugin -covariance true -endPlugin -export OUTPUT -runfork1
    run_pipeline.pl -Xmx150g -fork1 -importGuess SNPFILE -KinshipPlugin -method Centered_IBS -endPlugin -export OUTPUT -exportType SqrMatrix

The output file of GWAS were futher arranged to two types files, and these files were used as the input file of custom script(Code08) file to get candidate eQTLs
File1 :
    gene/sRNA_id    SNP_id  SNP_chr SNP_loci    p-value
File2 :
    SNP_id  SNP_loci-1  SNP_loci    gene/sRNA_id    p-value
Commandline : perl candidate_eQTL_find.pl File1 File2 > OUTPUT
the software bedtools was needed in this script.
    
    Code08.candidate_eQTL_find.pl (identify the candidate eQTLs). 
    
Then all candidate eQTLs were further classified to major eQTL region or satellite eQTL region according to the p-value of lead SNP or joint effect (when the p-value are equal) 

    Code09.LD_treatment.pl (filter the candidate eQTL throuth LD and p-value). 
    Code10.arrange_joint_effect.pl (arrange the joint effect result of candidate eQTLs). 
    Code11.major_eQTL.pl(get the major eQTL information). 
    Code12.SNPs_used_for_JE_calculation.pl (arrange the format of SNPs used to calculate the joint effects). 

For IR identification
    Firstly, The maize V4 genome sequence were split to series 100kb fregments
    
    Code13.SplitGenome.pl (split genome into the given lengths. The splited fragments will be further blast against itself by blastn at NCBI). 
    Code14.Treat_blastn_output.pl (arrange the output of the blast for further analyse).
    Code15.IRfinder.pl (identify the IR region and give the IR ratio). 


For Fst and XPCLR calculation
    To calculate the Fst value, the hmp format SNPfiles were transformed to VCF format by using 


For figure drawing
    
    Code16.LD_figure.pl (used to draw figure 2c). 
    Code17.Maize_V4_eQTL_map_drawing2.pl (used to draw figure 2a). 
    Code18.Maize_V4_eQTL_map_drawing.pl (used to draw Supplementary Results Fig. 4a).
    Code19.mhtplot.pl (used to draw Manhattan plots in Supplementary Results Fig. 3f, 3h). 
  
If there are any questions about using these scripts/code, please contact the authors at mingqiudai@mail.hzau.edu.cn.
